### ISSUE TITLE
[Noob or Bug] Assets Blob Filter

### ISSUE THREAD

**Issue by albertowd** [vercel/pkg#1146]
Hi, I did look out for the documentation, but I was unable to find an example. I need to filter assets per arch type, so the final executable does not have to pack all binaries dependencies.

My problem: the new `zeromq` lib now users prebuilt binaries, if available. Native node handles it very well, but it downloads all available binaries to the lib folder. When I package my executable I want to tell the `pkg` lib to filter only the current lib architecture so the output executable can be smaller.

Now:
```json
"pkg": {
  "assets": [
    "node_modules/zeromq/prebuild/win32-ia32/node.abi72.node",
    "node_modules/zeromq/prebuild/win32-ia32/node.abi83.node",
    "node_modules/zeromq/prebuild/win32-x64/node.abi72.node",
    "node_modules/zeromq/prebuild/win32-x64/node.abi83.node"
  ]
}
```

Example:
```json
"pkg": {
  "assets": [
    "node_modules/zeromq/prebuild/${PKG_OS}-${PKG_ARCH}/node.abi${PKG_NODE_MODULE_VERSION}.node"
  ]
}
```

PS.: I now that is a lib limitation because it does now respect the `__dirname` require. But would be nice to have that feature to work with.

**Comment by albertowd** [comment]
Just to write the `zeromq` package warning:

```powershell
Warning Cannot include directory %1 into executable.
  The directory must be distributed with executable as %2.
  %1: node_modules\zeromq\prebuilds
  %2: path-to-executable/prebuilds
```

**Comment by robertsLando** [comment]
You could just write a bash script that when you run pkg adds the required assets

**Comment by albertowd** [comment]
Yes, I can but that adds a layer of complexity when someone else pick up my work to do in the future. For me it's not complicated, my CD/CI is already functional but it's not as simple as call `npm run package:x86` or `npm run package:x64`. ^^

**Comment by CMCDragonkai** [comment]
> You could just write a bash script that when you run pkg adds the required assets

Can you explain what an example bash script like this would look like? How do you actually include those directories into the executable? I thought that with #837, pkg supports adding these files into the executable. But this warning still exists... so am I supposed to be distributing executable with these additional files thus necessitating a self-extracting archive?

**Comment by robertsLando** [comment]
>  I thought that with #837, pkg supports adding these files into the executable. But this warning still exists... so am I supposed to be distributing executable with these additional files thus necessitating a self-extracting archive?

When the files are in a `prebuilds` folder (for example projects that use prebuildify to build native addons) that is still not supported and so the warning is shown, what you have to do is to simply add this folders in `assets` on package.json, you will still see the warniing but the executable will include them

```js
"pkg": {
    "assets": [
      "/node_modules/@serialport/bindings/**",
      "/node_modules/leveldown/**"
    ],
  },
```
